torchgen/autoheuristic/gen_data_pad_mm.py

```diff
@@ -0,0 +1,208 @@
+# mypy: ignore-errors
+
+import argparse
+import random
+import time
+from typing import Any, Tuple
+
+import torch
+
+torch.set_default_device("cuda")
+from triton.testing import do_bench
+
+from torch._inductor.utils import fresh_inductor_cache
+
+# A100: 81920MiB
+# without a threshold we sometimes run out of memory
+threshold_memory = 85899345920 / 4
+
+# probability that a dimension is unaligned
+p_unaligned = 0.25
+
+# probability that a tensor is "prepadded", i.e. pad_mm excludes time it takes to pad from benchmarking
+p_prepadded = 0.2
+
+# probability that we pick from uniform distribution
+p_uniform = 0.5
+
+# probability that a tensor is transposed
+p_transposed = 0.1
```

Comment:
this seems low, the weight is almost always transposed

Reply:
I increased the probability that exactly one tensor is transposed.